compiler/rustc_hir/src/hir.rs

```diff
@@ -3064,6 +3064,27 @@ impl<'hir> Node<'hir> {
             Node::Crate(_) | Node::Visibility(_) => None,
         }
     }
+
+    /// Returns `Constness::Const` when this node is a const fn/impl.
+    pub fn constness(&self) -> Constness {
+        match self {
+            Node::Item(Item {
+                kind: ItemKind::Fn(FnSig { header: FnHeader { constness, .. }, .. }, ..),
+                ..
+            })
+            | Node::TraitItem(TraitItem {
+                kind: TraitItemKind::Fn(FnSig { header: FnHeader { constness, .. }, .. }, ..),
```

Comment:
this cannot happen (yet), maybe put it in a separate match arm and assert that the constness is `NotConst`

Reply:
I'll just revert this change since it is caught elsewhere.